### ISSUE TITLE
Responses gets mixed when used with parallel.foreach

### ISSUE THREAD

**Issue by montumodi** [restsharp/RestSharp#770]
I am not sure if rest sharp is thread safe or not but for some reason when i am running 10k requests in parallel, it is mixing the responses. 

e.g : 1 response is getting attached to multiple requests.

Any pointer?

**Comment by hallem** [comment]
i'm not sure honestly.  it depends on how you're calling things too.  i don't use the parallel foreach much so it's never come up for me.

**Comment by spamguy** [comment]
Same. I usually spawn one client per thread to work around this.

**Comment by alexeyzimarev** [comment]
Duplicate of #951
